packages/data/lib/repository/auth_repository.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:domain/model/firebase_user_email.dart';
+import 'package:domain/repository/auth_repository.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_facebook_auth/flutter_facebook_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+const _usersCollection = 'star_movie_users';
+
+class AuthRepositoryImpl implements AuthRepository {
+  final FirebaseAuth firebaseAuth;
```

Comment:
why do we need `FirebaseAuth`?

Reply:
we use his method "signInWithCredential", to throw our credits in firebase